### PR TITLE
feat(web): support restful api for split import project [VIZ-1508]

### PR DESCRIPTION
**Pull request by mkumbobeaty**
# Overview
In this Pr added restful API for split import project and remove the `importProject` mutation  from graphql.

## What I've done
- implement restful api setup for our system
- implement the split import project API 

## Summary by CodeRabbit

- **New Features**
  - Introduced a centralized provider for RESTful API communication, enabling secure and consistent API requests across the app.
- **Bug Fixes**
  - Improved file import reliability by switching to a chunked upload process via REST API, enhancing support for large project files.
- **Tests**
  - Added comprehensive tests for the new chunked project import functionality, covering success, failure, and edge cases.
- **Refactor**
  - Replaced the previous GraphQL-based project import with a RESTful, chunked upload approach.
- **Chores**
  - Updated provider structure to include the new RESTful API context.
  - Adjusted translation file order for consistency.
  - Renamed import handler for clarity in the project dashboard.
  - Renamed active task tracking hooks and state from GraphQL-specific to generic API task naming for clarity.

### CODE REVIEW

File: web/src/beta/utils/file.ts [hunk 1]
```diff
@@ -40,3 +41,30 @@ export async function fetchAndZipFiles(
 
   return file;
 }
+
+const CHUNK_SIZE = 16 * 1024 * 1024; // 16MB
+
+export const uploadFile = async (file: File, teamId: string, axios: any) => {
```
Comment (airslice) [inline web/src/beta/utils/file.ts:47]: Could you rename the function to be more specific? I think it's not a general file upload, it has a certain end point as `/split-import` so it's for split import project only.

Comment (airslice) [reply web/src/beta/utils/file.ts:47]: Or better rename & move this to projectApi.ts, api is abstracted, it could use gql or rest.

Comment (airslice) [reply web/src/beta/utils/file.ts:47]: Do we still need this function after you move it to projectApi?
If you want to make a general upload file with chunks function here please expose the formData property & endpoint & other configs as params of this function.

File: web/src/services/api/projectApi.ts [hunk 2]
```diff
@@ -533,47 +535,38 @@
...
           text: t("Failed to import project.")
         });
-        return { status: "error" };
+        console.log("Restful: Failed to import project", error);
       }
-    },
-    [importProjectMutation, t, setNotification]
-  );
+    }
+  };
```
[the remark this answers is not in the document]
Comment (airslice) [reply web/src/services/api/projectApi.ts:569]: Please read the comments and fix.
The fix provided by coderabbitai might be not good as well, i think we should consider it success only when all requests succeed.

[the remark this answers is not in the document]
Comment (airslice) [reply web/src/services/api/projectApi.ts:569]: Please test manually & try add unit test for this.

**Review by airslice — APPROVED** [review]

**Review by airslice — APPROVED** [review]
